creator-node/src/serviceRegistry.js

```diff
@@ -193,6 +195,72 @@ class ServiceRegistry {
     app.use('/health/bull', serverAdapter.getRouter())
   }
 
+  /**
+   * Wrap up current jobs in target queues under a max time. Used on death of app
+   */
+  async wrapUpQueueJobs() {
+    this.logInfo('Wrapping up jobs...')
+
+    const { queue: syncProcessingQueue } = this.syncQueue
+    const { queue: asyncProcessingQueue } = this.asyncProcessingQueue
+    const { queue: imageProcessingQueue } = this.imageProcessingQueue
+    const { queue: transcodingQueue } = this.transcodingQueue
+
+    const queues = [
+      syncProcessingQueue,
+      asyncProcessingQueue,
+      imageProcessingQueue,
+      transcodingQueue,
+      this.manualSyncQueue,
+      this.recurringSyncQueue,
+      this.stateReconciliationQueue
+    ]
+
+    const wrapUpCurrentJobs = async () => {
+      const responses = await Promise.all(
+        queues.map(async (queue) => {
+          try {
+            // Returns a promise that resolves when all jobs currently being processed by this worker have finished.
+            // https://github.com/OptimalBits/bull/blob/develop/REFERENCE.md#queuewhencurrentjobsfinished
+            await queue.whenCurrentJobsFinished()
```

Comment:
ah this sweet function

in the case of manual sync queue jobs tho, i think we actually want to wait for all jobs, not just the ones currently being processed - likely the way to do this would be to poll until queue size empty